crypto/rand_test.go

```diff
@@ -4,13 +4,24 @@ import (
 	"testing"
 )
 
-// TestRandIntnPanics tests that RandIntn panics if n <= 0.
-func TestRandIntnPanics(t *testing.T) {
+// panics returns true if the function fn panicked.
+func panics(fn func()) (panicked bool) {
 	defer func() {
-		if r := recover(); r == nil {
-			t.Error("expected panic for n <= 0")
-		}
+		panicked = (recover() != nil)
 	}()
-	RandIntn(0)
-	RandIntn(-1)
+	fn()
+	return
+}
```

Comment:
This is starting to go outside of the scope for this PR, but next PR can you move this function to the build package? `build.Panics`

Also, instead of returning an error you should return the panic message. `nil` means no panic, and then anything else is whatever object the panic threw. (panic throws an interface)

This pattern is used a lot in the `consensus` package, but without the helper function. No need to migrate it over proactively, but if you run into it feel free to update.